realm/realm-jni/src/io_realm_internal_tableview.cpp

```diff
@@ -25,20 +25,32 @@
 using namespace realm;
 
 // if you disable the validation, please remember to call sync_in_needed() 
-#define VIEW_VALID_AND_IN_SYNC(env, ptr) view_valid_and_in_sync(env, ptr)
+#define VIEW_VALID_AND_IN_SYNC(env, ptr) (is_view_valid(env, ptr) && is_view_in_sync(env, ptr))
 
-inline bool view_valid_and_in_sync(JNIEnv* env, jlong nativeViewPtr) {
-    bool valid = (TV(nativeViewPtr) != NULL);
-    if (valid) {
-        if (!TV(nativeViewPtr)->is_attached()) {
-            ThrowException(env, TableInvalid, "The Realm has been closed and is no longer accessible.");
-            return false;
-        }
-        TV(nativeViewPtr)->sync_if_needed();
+inline bool is_view_valid(JNIEnv* env, jlong nativeViewPtr) {
+    if (TV(nativeViewPtr) == NULL) {
+        // Should never get here
+        ThrowException(env, FatalError, "Null pointer of table view.");
+        return false;
+    }
+    if (!TV(nativeViewPtr)->is_attached()) {
+        ThrowException(env, TableInvalid, "The Realm has been closed and is no longer accessible.");
+        return false;
     }
-    return valid;
+    return true;
 }
 
+inline bool is_view_in_sync(JNIEnv* /*env*/, jlong nativeViewPtr) {
```

Comment:
The name suggests that it's a predicate but the side effect is that the table view is sync'ed.